src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java

```diff
@@ -76,19 +76,8 @@
 
     @Override
     protected void processFiltered(File file, FileText fileText) {
-        // Cannot use lines as the line separators have been removed!
         try {
-            final RandomAccessFile randomAccessFile = new RandomAccessFile(file, "r");
-            boolean threw = true;
-            try {
-                if (!endsWithNewline(randomAccessFile)) {
-                    log(0, MSG_KEY_NO_NEWLINE_EOF, file.getPath());
-                }
-                threw = false;
-            }
-            finally {
-                Closeables.close(randomAccessFile, threw);
-            }
+            readAndCheckFile(file);
```

Comment:
you need to move `randomAccessFile` out of TRY scope and merge TRYs, so `finally` become part of top TRY.